torch/_subclasses/fake_tensor.py

```diff
@@ -444,8 +453,25 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
             else:
                 return args[0].fake_device
 
+        with enable_torch_dispatch_mode(self), torch.overrides.enable_reentrant_dispatch():
+            if func in meta_table:
+                r = meta_table[func](*args, **kwargs)
+                return r
```

Comment:
Would be worth a comment here, along the lines of the fact that the Python meta function is more likely to support dynamic shapes, than the C++ one

Reply:
It's not clear to me why this is needed, since the defs in the meta table should be dispatched to anyway through meta registrations

Reply:
We want to error if we don't hit this table.